Sources/Application/SceneDelegate.swift

```diff
@@ -6,68 +6,75 @@ import Logging
+                             sessionManager: sessionManager)
+    }()
+
+    let analyticsService: AnalyticsService = GAnalytics()
+    private lazy var analyticsCenter = {
+        AnalyticsCenter(analyticsService: analyticsService,
+                        analyticsPreferenceStore: UserDefaultsPreferenceStore())
+    }()
+
+    var appBooted = false
+
     func scene(_ scene: UIScene,
                willConnectTo session: UISceneSession,
                options connectionOptions: UIScene.ConnectionOptions) {
         guard let windowScene = (scene as? UIWindowScene) else {
             fatalError("Window failed to initialise in SceneDelegate")
         }
-        windowManager = WindowManager(windowScene: windowScene)
+        // TODO: DCMAW-9866 | can we move this into the UI (viewDidAppear?) itself
+        trackSplashScreen()
+
+        rootCoordinator = QualifyingCoordinator(
+            window: UIWindow(windowScene: windowScene),
+            analyticsCenter: analyticsCenter,
+            appQualifyingService: appQualifyingService,
+            sessionManager: sessionManager,
+            networkClient: networkClient
+        )
+        rootCoordinator?.start()
+
         setUpBasicUI()
-        startMainCoordinator(windowManager: windowManager!)
     }
     
     func scene(_ scene: UIScene,
                continue userActivity: NSUserActivity) {
         guard let incomingURL = userActivity.webpageURL else { return }
-        coordinator?.handleUniversalLink(incomingURL)
-    }
-    
-    func startMainCoordinator(windowManager: WindowManagement) {
-        let tabController = UITabBarController()
-        let analyticsCenter = AnalyticsCenter(analyticsService: analyticsService,
-                                              analyticsPreferenceStore: UserDefaultsPreferenceStore())
-        coordinator = MainCoordinator(windowManager: windowManager,
-                                      root: tabController,
-                                      analyticsCenter: analyticsCenter,
-                                      networkClient: client,
-                                      sessionManager: sessionManager)
-        windowManager.appWindow.rootViewController = tabController
-        windowManager.appWindow.makeKeyAndVisible()
-        trackSplashScreen(analyticsCenter.analyticsService)
-        coordinator?.start()
+        rootCoordinator?.handleUniversalLink(incomingURL)
     }
     
     func sceneDidEnterBackground(_ scene: UIScene) {
-        if sessionManager.isSessionValid {
-            displayUnlockScreen()
-            shouldCallSceneWillEnterForeground = true
-        } else {
-            shouldCallSceneWillEnterForeground = false
-        }
+        rootCoordinator?.lock()
     }
     
     func sceneWillEnterForeground(_ scene: UIScene) {
-        if shouldCallSceneWillEnterForeground {
-            coordinator?.evaluateRevisit()
+        guard appBooted else {
+            appBooted = true
+            return
```

Comment:
Why do we need the guard if we're setting`appBooted` to true regardless?

Reply:
When the app is first booted this `sceneWillEnterForeground` is called but we don't want to call initiate on the qualifying service as this happens when the qualifying service is initialised.
Thinking about it now off the back of this question though, I don't think there's any reason _why_ we can't call it every time rather than doing it on initialisation 🙂
Will have a quick play around to see if there's any limitation to that but good shout!